### ISSUE TITLE
Outlook demo not working

### ISSUE THREAD

**Issue by markwilliams1234** [landrix/The-Drag-and-Drop-Component-Suite-for-Delphi#45]
I am running outlook in 64 bit and I have compiled the outlook demo in 64 bit (Delphi 10.4). When I try and drag an email into the app it is not accepted and the cursor icon never changes from the barred icon.

**Comment by landrix** [comment]
which demo?

Outlook or OutlookSource ?

**Comment by markwilliams1234** [comment]
Didn’t realise (until now) there were two.

 

I am having a problem with the Outlook demo, but have also tried the Outlook Source just now and same problem. However, I don’t think Outlook Source is set up to handle drag & drop direct from outlook.

 

__

**Comment by landrix** [comment]
you are right, mapi don't works currently under win64. i don't have any solution for this. It has nothing directly to do with drag and drop.

**Comment by markwilliams1234** [comment]
OK. Thanks.

 

__

**Comment by Ben-Silicon** [comment]
Hi, The problem is in the MAPIUtil.pas unit.
the headers are different if accessing with Win64bits application.
you have to remove the 'Historic Stack correction'
example : 
 HrGetOneProp:=GetProcAddress(MAPIDLLHandle,'HrGetOneProp@12');
must be replaced by 
 HrGetOneProp:=GetProcAddress(MAPIDLLHandle,'HrGetOneProp');

see this documentation : https://social.msdn.microsoft.com/Forums/vstudio/en-US/8d283d08-00fa-4e84-a96a-61b226f526e2/outlook-2010-mapi-headers?forum=vsto

See attached correction. [MAPIUtil.pas[]()](url
[MAPIUtil.pas.txt](https://github.com/landrix/The-Drag-and-Drop-Component-Suite-for-Delphi/files/6532545/MAPIUtil.pas.txt)
)

**Comment by markwilliams1234** [comment]
Hi,

 

Thanks for posting this fix and apologies for delay in responding. Will give this a try and report back if any issues.

 

Mark

 

__
